src/Microsoft.AspNetCore.Sockets.Http/HttpConnectionDispatcher.cs

```diff
@@ -110,9 +113,11 @@ public HttpConnectionDispatcher(ConnectionManager manager, ILoggerFactory logger
                     return;
                 }
 
+
                 _logger.EstablishedConnection(connection.ConnectionId, context.TraceIdentifier);
 
-                var ws = new WebSocketsTransport(options.WebSockets, connection.Application, connection.ConnectionId, _loggerFactory);
+                connection.TransportCapabilities = TransferMode.Binary | TransferMode.Text;
```

Comment:
EnsureConnectionStateAsync is called for all transports, we should just set it there and override it where it needs to be different.